RecoBTag/DeepFlavour/interface/deep_helpers.h

```diff
@@ -0,0 +1,108 @@
+#ifndef RecoBTag_DeepFlavour_deep_helpers_h
+#define RecoBTag_DeepFlavour_deep_helpers_h
+
+#include "DataFormats/GeometryCommonDetAlgo/interface/Measurement1D.h"
+#include "DataFormats/VertexReco/interface/VertexFwd.h"
+#include "TrackingTools/IPTools/interface/IPTools.h"
+
+
+
+namespace deep {
+
+  // adapted from DeepNTuples
+  inline const float& catch_infs(const float& in,
+                                 const float& replace_value) {
+    if(in==in){ // why is this ?
+      if(std::isinf(in))
+        return replace_value;
+      else if(in < -1e32 || in > 1e32)
+        return replace_value;
+      return in;
+    }
+    return replace_value;
+  } 
+
+  // adapted from DeepNTuples
+  inline const float catch_infs_and_bound(const float& in,
+                                          const float& replace_value,
+                                          const float& lowerbound,
+                                          const float& upperbound,
+                                          const float offset=0.,
+                                          const bool& use_offsets = true){
```

Comment:
passing by values is simpler than by ref (address size is larger than the value itself)

Reply:
sure, will change

Reply:
changed in https://github.com/cms-sw/cmssw/pull/19893/commits/1d9e7d0e37beede555eb8e48850dfbcf4141bcb0